app/scripts/background.js

```diff
@@ -470,18 +470,21 @@ function setupController(initState, initLangCode) {
     updateBadge,
   );
 
+  // todo: task plan to fix this for MV3, https://github.com/MetaMask/metamask-extension/issues/13926
   /**
    * Updates the Web Extension's "badge" number, on the little fox in the toolbar.
    * The number reflects the current number of pending transactions or message signatures needing user approval.
    */
   function updateBadge() {
-    let label = '';
-    const count = getUnapprovedTransactionCount();
-    if (count) {
-      label = String(count);
+    if (!process.env.ENABLE_MV3) {
+      let label = '';
+      const count = getUnapprovedTransactionCount();
+      if (count) {
+        label = String(count);
+      }
+      browser.browserAction.setBadgeText({ text: label });
+      browser.browserAction.setBadgeBackgroundColor({ color: '#037DD6' });
```

Comment:
```suggestion
      browser.action.setBadgeText({ text: label });
      browser.action.setBadgeBackgroundColor({ color: '#037DD6' });
```

Reply:
@adonesky1 : will this change work in MV3 also ?

Reply:
Yes these are changes only for MV3: https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#action-api-unification

I wasn't able to get this build to work properly without them...?